src/uu/top/src/top.rs

```diff
@@ -0,0 +1,275 @@
+        .collect::<Vec<_>>();
+
+    let filter = construct_filter(settings);
+
+    pids.into_iter()
+        .filter(|pid| filter(*pid))
+        .map(|it| {
+            pickers
+                .iter()
+                .map(move |picker| picker(it))
+                .collect::<Vec<_>>()
+        })
+        .collect()
+}
+
+/// Constructing filter from `Settings`
+fn construct_filter(settings: &Settings) -> Box<dyn Fn(u32) -> bool> {
+    let Some(ref filter) = settings.filter else {
+        return Box::new(|_: u32| true);
+    };
+
+    fn helper(f: impl Fn(u32) -> bool + 'static) -> Box<dyn Fn(u32) -> bool> {
+        Box::new(f)
+    }
+
+    match filter {
+        Filter::Pid(pids) => {
+            let pids = pids.clone();
+            helper(move |pid: u32| pids.contains(&pid))
+        }
+
+        Filter::User(user) => {
+            let user = user.to_owned();
+
+            helper(move |pid| {
+                let binding = sysinfo().read().unwrap();
+                let Some(proc) = binding.process(Pid::from_u32(pid)) else {
+                    return false;
+                };
+
+                let Some(uid) = proc.user_id() else {
+                    return false;
+                };
+
+                Users::new_with_refreshed_list()
+                    .get_user_by_id(uid)
+                    .map(|it| it.name())
+                    .unwrap_or("?")
+                    == user
+            })
+        }
+        // TODO: Implemented
+        Filter::EUser(_) => helper(|_| true),
+    }
+}
+
+#[allow(clippy::cognitive_complexity)]
+pub fn uu_app() -> Command {
+    Command::new(uucore::util_name())
+        .version(crate_version!())
+        .about(ABOUT)
+        .override_usage(format_usage(USAGE))
+        .infer_long_args(true)
+        .args([
+            // arg!(-b  --"batch-mode"                         "run in non-interactive batch mode"),
+            // arg!(-c  --"cmdline-toggle"                     "reverse last remembered 'c' state"),
+            // arg!(-d  --delay                <SECS>          "iterative delay as SECS [.TENTHS]"),
+            // arg!(-E  --"scale-summary-mem"  <SCALE>         "set mem as: k,m,g,t,p,e for SCALE"),
+            // arg!(-e  --"scale-task-mem"     <SCALE>         "set mem with: k,m,g,t,p for SCALE"),
+            // arg!(-H  --"threads-show"                       "show tasks plus all their threads"),
+            // arg!(-i  --"idle-toggle"                        "reverse last remembered 'i' state"),
+            // arg!(-n  --iterations           <NUMBER>        "exit on maximum iterations NUMBER"),
+            arg!(-O  --"list-fields"                        "output all field names, then exit"),
+            // arg!(-o  --"sort-override"      <FIELD>         "force sorting on this named FIELD"),
+            arg!(-p  --pid                  <PIDLIST>       "monitor only the tasks in PIDLIST")
+                .action(ArgAction::Append),
```

Comment:
I think a value parser is missing here. When running `cargo run -q top -p <PID>` I get the following error:
```
thread 'main' panicked at src/uu/top/src/top.rs:41:14:
Mismatch between definition and access of `pid`. Could not downcast to u32, need to downcast to alloc::string::String
```

Reply:
Fixed 

```shell
❯ cargo run -q top -p 1
TODO


 PID  USER  PR  NI    VIRT  RES   SHR   S  %CPU  %MEM  TIME+    COMMAND 
 1    root  0   TODO  TODO  TODO  TODO  S  0.00  0.0   2:19.31  /sbin/init splash 
```